### PR TITLE
2deteCT dataset: pre_processing function and dataset class

### PR DESCRIPTION
**Pull request by Emvlt**

### CODE REVIEW

**Comment by AnderBiguri** [comment]
yay!

Can I commit to this branch? I would like to make it "LION-like", i.e. change how parameters are passed etc.

**Comment by Emvlt** [comment]
Go ahead for the lionisation :D

File: LION/data_loaders/deteCT.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        ### Defining the path to scan data
+        self.path_to_scan_data = self.path_to_dataset.joinpath("scan_settings.json")
+        ### Defining the path to the data record
+        if default:
+            self.path_to_data_record = self.path_to_dataset.joinpath(
+                f"default_data_records.csv"
+            )
+        else:
+            self.path_to_data_record = self.path_to_dataset.joinpath(
+                f"complete_data_records.csv"
```
Comment (AnderBiguri) [inline LION/data_loaders/deteCT.py:61]: Can you explain what this does?

**Comment by AnderBiguri** [comment]
TODO by me:

- [ ] Parametrize inputs
- [ ] Change parameter "mode" name to something else (`mode` is used for training/validation/test in other data loaders)
- [ ] Add Validation

File: LION/data_loaders/deteCT.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        """
+        self.transforms = transforms
+
+        ### We query the dataset subset
+        self.slice_dataframe: pd.DataFrame
+        """
+        The slice_dataframe (pd.Dataframe) is the subset of interest of the dataset.
+        The self.data_record argument becomes the slice_dataframe once we have queried it
+        Example of query: 'detector==1'
+        If the no query argument is passed, data_record_subset == data_record
```
Comment (AnderBiguri) [inline LION/data_loaders/deteCT.py:119]: The rest of the documentation is excelent, but I understood nothing of this....

File: LION/data_loaders/deteCT.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        self.train = train
+        """
+        The train (bool) argument defines if the dataset is used for training or testing
+        """
+        self.training_proportion = training_proportion
+        # Defining the train proportion
+        """
+        The training_proportion (float) argument defines the proportion of the training dataset used for training
+        """
+        self.transforms = transforms
```
Comment (AnderBiguri) [inline LION/data_loaders/deteCT.py:111]: What is this?

File: LION/data_loaders/deteCT.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        unique_identifiers = self.slice_dataframe["sample_index"].unique()
+        record = {"sample_index": [], "first_slice": [], "last_slice": []}
+        for identifier in unique_identifiers:
+            record["sample_index"].append(identifier)
+            subset = self.slice_dataframe[
+                self.slice_dataframe["sample_index"] == identifier
+            ]
+            record["first_slice"].append(subset["first_slice"].iloc[0])
+            record["last_slice"].append(subset["last_slice"].iloc[0])
+        self.sample_dataframe = pd.DataFrame.from_dict(record)
```
Comment (AnderBiguri) [inline LION/data_loaders/deteCT.py:159]: I don't get this one either, likely due to my lack of Pandas knowledge.

**Comment by AnderBiguri** [comment]
@Emvlt  please careful at commiting, you just added +5000 lines of the egg.info, which should have been .gitignore'd . 
I think the best would be to undo the commits, rather than remove them, otherwise you will clog the git history.

**Comment by AnderBiguri** [comment]
Notes with Max:

- [ ] Only have default records
- [ ] have flags for FF, DF, log, default all true. Warning if FF/DF not available and detector query empty
- [ ] query default empty
- [ ] detector query
- [ ] Advanced query: sample mix. -> fix records to reflect this (and remove error)

**Comment by AnderBiguri** [comment]
This is essentially functional, and verified with the recon that we can provide an operator. 

There is only a couple of thigns missing (see above checks), plus:

- [ ] The operator does not match 100% the recon, because of a half-pixel shift of where the center of pixels are defined w.r.t. the original recon. Trying to see if this can be done by defining a small offset in tomosipo operators (https://github.com/ahendriksen/tomosipo/issues/28), but otherwise the recon will need to be done in 2048, then downsample.

**Comment by AnderBiguri** [comment]
Finished!!!!!

**Pull request merged**